runtime/vm/createramclass.cpp

```diff
@@ -1944,16 +1944,31 @@ loadFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *class
 	UDATA flattenableFieldCount = 0;
 	bool eligibleForFastSubstitutability = true;
 
+	/* Preload classes marked for preloading */
+	if (J9_ARE_ALL_BITS_SET(romClass->optionalFlags, J9_ROMCLASS_OPTINFO_PRELOAD_ATTRIBUTE)) {
+		U_32 *numberOfPreloadClassesPtr = getNumberOfPreloadClassesPtr(romClass);
+
+		for (U_32 i = 0; i < *numberOfPreloadClassesPtr; i++) {
+			J9UTF8 *preloadClassNameUtf8 = preloadClassNameAtIndex(numberOfPreloadClassesPtr, i);
+
+			U_8 *preloadClassName = J9UTF8_DATA(preloadClassNameUtf8);
+			U_16 preloadClassLength = J9UTF8_LENGTH(preloadClassNameUtf8);
+
+			internalFindClassUTF8(currentThread, preloadClassName, preloadClassLength, classLoader, classPreloadFlags);
```

Comment:
Is this all you meant to do here? This looks incomplete. My understanding from the issue is that it will replace the Q case.

Reply:
That is all I meant to do, since I believe for now we're keeping in both preloading cases. @hangshao0 can you confirm that?

Reply:
Eventually we will replace all Q type case with the preload attributes. However, OpenJDK has not updated javac to generate this attribute for all Q types yet, so we need to keep the Q type case now for things to function correctly.

Reply:
Thanks for the clarification.